### ISSUE TITLE
Datasim is requiring optional features

### ISSUE THREAD

**Issue by FlorianTolk** [yetanalytics/datasim#55]
While running a profile test, I noticed datasim is requiring 2 optional properties.
1 - When defining the profile, it expects the "seeAlso" property, an optional property at the [profile definition level](https://github.com/adlnet/xapi-profiles/blob/master/xapi-profiles-structure.md#60-profile-properties).
2 - In all primary [patterns](https://github.com/adlnet/xapi-profiles/blob/master/xapi-profiles-structure.md#patterns), it requires a definition object.

**Comment by kelvinqian00** [comment]
The first point is something we'll look into, but regarding the second point, that is intentional. The xAPI Profile spec says (below the table in section 9):
> A primary Pattern MUST include `prefLabel` and `definition`. They are optional otherwise.

**Comment by FlorianTolk** [comment]
My bad, I completely missed this in the spec

**Comment by kelvinqian00** [comment]
Anyways I believe that the first issue is caused by `nil` values in the Profile properties. For example, executing the following line:
```
(com.yetanalytics.datasim.input/from-location :profile :json "dev-resources/profiles/minimal.jsonld)
```
will return the following Profile:
```clojure
{:id "https://xapinet.org/xapi/yet/minimal",
 :_context "https://w3id.org/xapi/profiles/context",
 :conformsTo "https://w3id.org/xapi/profiles#1.0",
 :prefLabel {:en "Minimal - Experimental xAPI Profile"},
 :definition {:en "This xAPI Profile demonstrates the minimal required properties of an xAPI profile"},
 :seeAlso nil,
 :versions [{:id "https://xapinet.org/xapi/yet/minimal/v1", :generatedAtTime "2020-03-25T15:45:31.907Z"}],
 :author {:url "https://www.yetanalytics.com/", :name "Yet Analytics", :type "Organization"},
 :concepts nil,
 :templates nil,
 :patterns nil,
 :type "Profile"}
```
Notice the `nil` values for `:seeAlso`, `:concepts`, `:templates`, and `:patterns`, which would fail validation since nil/null values are not allowed. Is this indeed the issue you are facing?

**Comment by FlorianTolk** [comment]
These properties did not even exist in the profile I was working with, and I was getting this error.
It's an easy workaround to just add them though

**Comment by kelvinqian00** [comment]
Yes, if (for example) the `seeAlso` property did not exist in your profile, then calling `from-location` on it would result in the property being assigned a `nil` value internally. So it seems you are experiencing exactly what I just described.

Also for reference, here is the original `dev-resources/profiles/minimal.jsonld` file:
```json
{
  "id" : "https://xapinet.org/xapi/yet/minimal",
  "definition" : {
    "en" : "This xAPI Profile demonstrates the minimal required properties of an xAPI profile"
  },
  "@context" : "https://w3id.org/xapi/profiles/context",
  "prefLabel" : {
    "en" : "Minimal - Experimental xAPI Profile"
  },
  "type" : "Profile",
  "author" : {
    "url" : "https://www.yetanalytics.com/",
    "name" : "Yet Analytics",
    "type" : "Organization"
  },
  "conformsTo" : "https://w3id.org/xapi/profiles#1.0",
  "versions" : [
    {
      "id" : "https://xapinet.org/xapi/yet/minimal/v1",
      "generatedAtTime" : "2020-03-25T15:45:31.907Z"
    }
  ]
}
```
Notice that the minimal Profile does not have any of the optional properties.

**Comment by kelvinqian00** [comment]
Try 27ae7a435dc23e98abf6479d0df2a03d333595b4 - I removed the optional properties from the internal Profile defrecord, so you should no longer get `nil` values.

**Comment by kelvinqian00** [comment]
This has been fixed in PR #57.
